### PR TITLE
Fix changing an IndicatorItemView's size value.

### PR DESCRIPTION
**Pull request by DashTheDev**
### Details
Due to the order in which style properties are applied to elements, along with issues with the **Frame** element, there's conflicts with the Size value being set on the **IndicatorItemView**. After testing, I found when the view is rendered, style will apply a default size value on the **CircleFrame** and then the constructor of the **IndicatorItemView** was overriding the size.

Due to this, if anyone was to set the size of an **IndicatorItemView** using the selected and unselected styles, the new size was not respected. For now, I've moved the default "set" of size to the BindableProperty creation and this seems to fix said issues.

However, the fact that the **IndicatorItemView** inherits a **Frame** causes the effective minimum size to be the following:

image
_Code implemented on the base **Frame** element._

This default value creator really causes issues with the whole order of style applied values, constructors, etc. It also behaves completely differently on iOS and Android. Android will not allow heights/widths smaller than the padding unless you set MinimumHeightRequest/MinimumWidthRequest but iOS doesn't care and will ignore the padding.

To address all of these issues further and in a way that will align with user expectations, please see my proposed changes for a future pull request that I'll make if advised/approved of.

### Tested Platforms
- iOS
- Android
- Windows

### Future Proposed Changes
Since the following proposal will alter **IndicatorItemView** pretty heavily and most likely be a breaking change/major update for end users, please read my suggestion and advise if I'm allowed to flesh out and develop said changes. If you're not keen on me making these changes, all good! Forget I suggested this :)

I propose we change **CircleFrame** to inherit from **Border** rather than **Frame** (so a **CircleBorder**). With this change, we'll need to update the docs and README but the benefits will only improve user experience. Making the switch will address the following:

1. Align with Microsoft [recommendations](https://learn.microsoft.com/en-us/dotnet/maui/user-interface/controls/frame?view=net-maui-8.0&viewFallbackFrom=net-maui-7.0):

> The [Frame](https://learn.microsoft.com/en-us/dotnet/api/microsoft.maui.controls.frame) class existed in Xamarin.Forms and is present in .NET MAUI for users who are migrating their apps from Xamarin.Forms to .NET MAUI. If you're building a new .NET MAUI app it's recommended to use [Border](https://learn.microsoft.com/en-us/dotnet/api/microsoft.maui.controls.border) instead, and to set shadows using the Shadow bindable property on [VisualElement](https://learn.microsoft.com/en-us/dotnet/api/microsoft.maui.controls.visualelement). For more information, see [Border](https://learn.microsoft.com/en-us/dotnet/maui/user-interface/controls/border?view=net-maui-8.0) and [Shadow](https://learn.microsoft.com/en-us/dotnet/maui/user-interface/shadow?view=net-maui-8.0).

2. Allow complete control for users over the sizing of **IndicatorItemViews** without the strange default values, overhead (like shadow) and shenanigans introduced by the **Frame** element.

3. Remove the need for a CornerRadius property, as **Border** allows us to set StrokeShape as an ellipse.

### CODE REVIEW

**Comment by AndreiMisiukevich** [comment]
Hi man! Thank you so much for your kind contributions!

Could you please confirm that these changes don’t bring any breaking changes (so the indicators control still works the same way without any additional setup, like <IndicatorsControl />)

**Review by AndreiMisiukevich — APPROVED** [review]

**Comment by DashTheDev** [comment]
Of course, happy to help! This library has been great for a project I'm working on and just want to help improve some of the things I ran into :)

If you're referring to this PR (which I'm sure you're not), everything runs the same before and after on all three tested platforms.

As for the proposed future changes, the only immediate issue I see for end users is dependent on if they rely on the base **Frame** element properties on the **IndicatorItemView** at all. As per the indicator styling section on the README, users are shown an example to style the indicators like so:

```XAML
<ContentPage.Resources>
    <ResourceDictionary>
        <Style x:Key="ActiveIndicator" TargetType="Frame">
            <Setter Property="BackgroundColor" Value="Red" />
        </Style>
        <Style x:Key="InactiveIndicator" TargetType="Frame">
            <Setter Property="BackgroundColor" Value="Transparent" />
            <Setter Property="OutlineColor" Value="Red" />
        </Style>
    </ResourceDictionary>
</ContentPage.Resources>

... 

<controls:IndicatorsControl ToFadeDuration="1500"
          SelectedIndicatorStyle="{StaticResource ActiveIndicator}"
          UnselectedIndicatorStyle="{StaticResource InactiveIndicator}"/>
```
_As a side note, the OutlineColor property doesn't exist on Frame in MAUI. It's now BorderColor. So we should probably update this example regardless. If I refer to the OutlineColor, I am referring to MAUI's BorderColor on **Frame**._

So I did some initial testing of the **Frame** to **Border** on iOS, Android and Windows. If the base type of **IndicatorItemView** was to change, this example styling technically would still work surprisingly (at least it did on the three platforms I tested). It just won't apply any properties that aren't found on the **Border**, so the BackgroundColor would apply as normal but the OutlineColor would not be applied (but would not error, which is cool).

However, if a user had a style setup like the following:

```XAML
<ContentPage.Resources>
    <ResourceDictionary>
        <Style x:Key="SelectedIndicator" TargetType="controls:IndicatorItemView">
            <Setter Property="HasShadow" Value="False" />
        </Style>
    </ResourceDictionary>
</ContentPage.Resources>
```

Then they would have a compile error since HasShadow doesn't exist on the base **Border** element, but it did on the **Frame** element. Obviously the same goes for if they were trying to set any properties that don't exist on the **IndicatorItemView** anymore due to the change.

After testing with the preliminary **Frame** to **Border** changes I made, all examples work out of the box on Windows, iOS and Android.

TLDR: If they had everything setup out of the box and/or in line with even the current examples then in theory if my tests weren't specific to my machine, then everything would work as per usual. But, if they do have custom setups similar to the previous example, they may face some grief depending on their setup.

Thoughts?

**Comment by AndreiMisiukevich** [comment]
Awesome, thanks you!
